tests/e2e/vsphere.go

```diff
@@ -89,6 +90,59 @@ func (vs *vSphere) queryCNSVolumeWithResult(fcdID string) (*cnstypes.CnsQueryRes
 	return &res.Returnval, nil
 }
 
+// queryCNSVolumeSnapshotWithResult Call CnsQuerySnapshots
+// and returns CnsSnapshotQueryResult to client
+func (vs *vSphere) queryCNSVolumeSnapshotWithResult(fcdID string,
+	snapshotId string) (*cnstypes.CnsSnapshotQueryResult, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	// Connect to VC
+	connect(ctx, vs)
+	var snapshotSpec []cnstypes.CnsSnapshotQuerySpec
+	snapshotSpec = append(snapshotSpec, cnstypes.CnsSnapshotQuerySpec{
+		VolumeId: cnstypes.CnsVolumeId{
+			Id: fcdID,
+		},
+		SnapshotId: &cnstypes.CnsSnapshotId{
+			Id: snapshotId,
+		},
+	})
+
+	queryFilter := cnstypes.CnsSnapshotQueryFilter{
+		SnapshotQuerySpecs: snapshotSpec,
+		Cursor: &cnstypes.CnsCursor{
+			Offset: 0,
+			Limit:  100,
+		},
+	}
+
+	req := cnstypes.CnsQuerySnapshots{
+		This:                cnsVolumeManagerInstance,
+		SnapshotQueryFilter: queryFilter,
+	}
+
+	err := connectCns(ctx, vs)
```

Comment:
do we have to do it for every call? just once for a run should be fine right?

Reply:
yes - removed the un-necessary call

Reply:
I still see it in the latest diff, can you check again